### PR TITLE
[PATCH v2] fdserver: handle signal interruption in connect()

### PR DESCRIPTION
**Pull request by joseppc**
Signed-off-by: Josep Puigdemont <josep.puigdemont@linaro.org>

### CODE REVIEW

**Comment by Bill-Fischofer-Linaro** [comment]
@JannePeltonen Can you review this and comment on whether you believe this is helpful for [Bug 3690](https://bugs.linaro.org/show_bug.cgi?id=3690)?

File: platform/linux-generic/odp_fdserver.c [hunk 1]
```diff
@@ -265,7 +265,9 @@ static int get_socket(void)
 	remote.sun_family = AF_UNIX;
 	strcpy(remote.sun_path, sockpath);
 	len = strlen(remote.sun_path) + sizeof(remote.sun_family);
-	if (connect(s_sock, (struct sockaddr *)&remote, len) == -1) {
+	while (connect(s_sock, (struct sockaddr *)&remote, len) == -1) {
+		if (errno == EINTR)
+			continue;
 		ODP_ERR("cannot connect to server: %s\n", strerror(errno));
```
Comment (joseppc) [inline platform/linux-generic/odp_fdserver.c]: I was thinking if we should account for other errors, like `ETIMEDOUT` or `ECONNREFUSED` (the server's backlog is set to 2 and could happen in busy situations).
These are unlikely events, of course, but they may need handling nonetheless.

**Comment by joseppc** [comment]
@JannePeltonen @Bill-Fischofer-Linaro I forgot to mention that this call runs on the context of ODP threads. I think it is safe enough to merge it in, but the question of handling other errors remains.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**